src/Eval.ml

```diff
@@ -68,13 +71,27 @@ let rec of_list = function
 
 let int2_fun f = int_fun (fun x -> int_fun (f x))
 
+let int_unop  op = int_fun  (fun x -> VNum (op x))
 let int_binop op = int2_fun (fun x y -> VNum (op x y))
 let int_cmpop op = int2_fun (fun x y -> of_bool (op x y))
 
+let int64_fun f = VFn (fun v cont ->
+  match v with
+  | VNum64 n -> cont (f n)
+  | _ -> failwith "Runtime error!")
+
+let int64'2_fun f = int64_fun (fun x -> int64_fun (f x))
```

Comment:
I don't like the name of this function. Maybe `int64_fun2` would be better? Then of course, we should rename `int2_fun` to `int_fun2`.